src/posix/platform/radio.cpp

```diff
@@ -146,61 +141,71 @@ void Radio::Init(void)
         SuccessOrDie(otPlatRadioSetRegion(gInstance, regionCode));
     }
 
-#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
-    maxPowerTable = mRadioUrl.GetValue("max-power-table");
-    if (maxPowerTable != nullptr)
+    ProcessMaxPowerTable(aRadioUrl);
+
+#if OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
     {
-        constexpr int8_t kPowerDefault = 30; // Default power 1 watt (30 dBm).
-        const char      *str           = nullptr;
-        uint8_t          channel       = ot::Radio::kChannelMin;
-        int8_t           power         = kPowerDefault;
-        otError          error;
-
-        for (str = strtok(const_cast<char *>(maxPowerTable), ","); str != nullptr && channel <= ot::Radio::kChannelMax;
-             str = strtok(nullptr, ","))
+        const char *enableCoex = aRadioUrl.GetValue("enable-coex");
+        if (enableCoex != nullptr)
         {
-            power = static_cast<int8_t>(strtol(str, nullptr, 0));
-            error = sRadioSpinel.SetChannelMaxTransmitPower(channel, power);
-            if (error != OT_ERROR_NONE && error != OT_ERROR_NOT_IMPLEMENTED)
-            {
-                DieNow(OT_ERROR_FAILED);
-            }
-            else if (error == OT_ERROR_NOT_IMPLEMENTED)
-            {
-                otLogWarnPlat("The RCP doesn't support setting the max transmit power");
-            }
-
-            ++channel;
+            SuccessOrDie(sRadioSpinel.SetCoexEnabled(enableCoex[0] != '0'));
         }
+    }
+#endif // OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
+}
+
+void Radio::ProcessMaxPowerTable(const RadioUrl &aRadioUrl)
+{
+    OT_UNUSED_VARIABLE(aRadioUrl);
 
-        // Use the last power if omitted.
-        while (channel <= ot::Radio::kChannelMax)
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    otError          error;
+    constexpr int8_t kPowerDefault = 30; // Default power 1 watt (30 dBm).
+    const char      *str           = nullptr;
+    uint8_t          channel       = ot::Radio::kChannelMin;
+    int8_t           power         = kPowerDefault;
+    const char      *maxPowerTable;
+
+    VerifyOrExit((maxPowerTable = aRadioUrl.GetValue("max-power-table")) != nullptr);
+
+    for (str = strtok(const_cast<char *>(maxPowerTable), ","); str != nullptr && channel <= ot::Radio::kChannelMax;
+         str = strtok(nullptr, ","))
+    {
+        power = static_cast<int8_t>(strtol(str, nullptr, 0));
+        error = sRadioSpinel.SetChannelMaxTransmitPower(channel, power);
+        if (error != OT_ERROR_NONE && error != OT_ERROR_NOT_IMPLEMENTED)
```

Comment:
nit: use `VerifyOrDie`?

Reply:
I have updated the code and optimized the two loops.